lib/source-map.js

```diff
@@ -101,10 +101,28 @@ SourceMap.prototype.addFileSource = function(filename, source, inputSrcMap) {
     inputSrcMap = this._resolveSourcemap(filename, url);
   }
 
+  var valid = true;
+
   if (inputSrcMap) {
-    source = this._addMap(filename, inputSrcMap, source);
+    try {
+      // TODO: don't stringify here
+      validator(source, JSON.stringify(inputSrcMap));
```

Comment:
This isn't ideal, future work can implement this more efficiently. At the very least, we have good tests now, future work to fix can use that as a foundation.